### ISSUE TITLE
Data too large to derive results within given time

### ISSUE THREAD

**Issue by TimmermanIDS** [RUG-IDS/Course-2017#21]
We've been working on the last.fm dataset but we run into the problem that the dataset is far too complex to allow us to derive results. We've already tried to only consider parts of the data where beethoven is present (Question 2.2), but this either results in incredibly biased results or no decent results at all. We've also tried to work with only a random sample of the complete dataset, but since the support of Beethoven is only about 1%, the support needs to be even lower than that  to get an itemset where beethoven is found together with another artist. However, the computation time explodes if you lower the minimum support. Reducing the amount of sample users is not possible as well because Beethoven is so infrequent that we need at least a thousand sample users to have ten people who listen to Beethoven.

What advice can you give us?

**Comment by mircealungu** [comment]
What language and what library are you working with?

**Comment by mircealungu** [comment]
I've had not much problems with R and a sample of 10% of the records in the DB

**Comment by rm-bwana** [comment]
Just to follow up on that. Although we allow the choice between the 3 languages, not all algorithms are implemented equally in all 3. Unfortunately the python algorithm seems to be I/O intensive which slows things down quite a bit. You may have to consider another choice if you are not satisfied with the speed at which it runs. 

As for the amount of data to be used, it is impossible to use the entire data of 2+GB. Therefore we recommend using some form of sampling (there are more than just simple random sampling) to reduce the size of the data set in use. You may consider Beethoven to be a underrepresented class of interest in such a scenario. Please make sure how you do this is **clearly stated** in your report so that we can follow the steps you took.

Finally, there will not be an ideal perfect answer that you can find. Just degrees of (in-)accuracy you are ready to accept. Unfortunately that is Data Science in a nutshell. We are not expecting you to find everything that could be interesting however we just want to see that you made the decision and can justify why you made it in a manner that makes sense.

**Comment by Daemonstool** [comment]
After attempting all methods (R, Python (Apyori and Apriori), Matlab), we came to the conclusion that R is definitely the way to go. We were able to parse the entire dataset with a support of 0.001 in matters of seconds after transforming the data into a sparse matrix. It would be appreciated to warn the students about the runtime differences of the different implementations as we tried to get the python version to work for more than 3 days :(. Especially since the Python packages are listed as a recommendation. Obviously we expected some runtime differences between the different versions, but we did not expect this to be significantly different.  We hope to have enough time to do the entire analysis before the deadline.  

Due to the issues that we encountered, we do request an extension of the deadline. This allows us and everyone who also ran into troubles with the Python and Matlab packages to do an extensive analysis. 

We also attempted to oversample Beethoven but this can lead to biased results. In the end we are happy that we can do the analysis on the whole dataset. 

Kind regards,
Team-9
